lib/elastic_apm/traceparent.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  # @api private
+  class Traceparent
+    class InvalidTraceparentHeader < StandardError; end
+
+    VERSION = '00'
+    HEX_REGEX = /[^[:xdigit:]]/
+
+    def initialize
+      @version = VERSION
+    end
+
+    def self.from_transaction(transaction)
+      new.tap do |t|
+        t.trace_id = SecureRandom.hex(16)
+        t.span_id = transaction.id
```

Comment:
Why do you set the `span_id` to `transaction.id`? From what I understand it should be set to the `span.id` that initiated the outgoing call to another service.